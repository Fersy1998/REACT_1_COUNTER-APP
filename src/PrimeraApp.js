import React from 'react';
//import React, {fragment} from 'react';
import PropTypes from 'prop-types';

const PrimeraApp=({ saludo, subtitulo="Soy un subtÃ­tulo"})=>{
    const saludoApp =  saludo;
   // console.log(props);
    const persona={
        nombre: "Rosaria",
        apellido:"Yvgnir",
        edad:34
    }
    
    return (   
    <>
    <h1><pre>{/*JSON.stringify(persona, null, 3)*/}{saludoApp}</pre></h1>
    {/*<p>Primera App</p>*/}
    !!!
    <p>{subtitulo}</p>
    </>
    )
}

PrimeraApp.propTypes={
   // props: PropTypes.string.isRequired,
   saludo: PropTypes.string.isRequired,
   subtitulo: PropTypes.string.isRequired,
}
/*
PrimeraApp.defaultProps={
    parrafo: "Soy un parrafo",

}*/
export default PrimeraApp;