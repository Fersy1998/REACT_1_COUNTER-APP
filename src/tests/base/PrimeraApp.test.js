import React from 'react';


//import {createRoot} from 'react-dom/client';
//import { getByAltText, getByText, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import PrimeraApp from '../../PrimeraApp';
import {shallow} from 'enzyme';
describe('pruebas en la primera APp.js', ()=>{
    /*
    test('should say if App is lacking something! "Hello, I´m Fersy"', () => {
        const saludo="Hello, I´m Fersy";
        const subtitulo="New One";
        //const wrapper= render(<PrimeraApp saludo={saludo} subtitulo="subtitulo"/>)
        const {getByText}=render(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />)
        console.log(getByText);
        expect(getByText(saludo)).toBeInTheDocument();
    
    
    
    })*/
    test('should say if App is lacking something new! "Hello, I´m Fersy"', () => {
        const saludo="Hello, I´m Fersy 23";
        const subtitulo="New One for third time";
        const wrapper=shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);
        console.log(wrapper);
        
        //const divRoot=createRoot(document.querySelector('#root'));
        //divRoot.render(wrapper);

        expect(wrapper).toMatchSnapshot();
    
    })
    
    test('should see the <p></p>', () => { 
        const saludo="Hello, I´m Fersy 23";
        const subtitulo="New One for third time";
        const wrapper=shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo}/>);
        console.log(wrapper);
        const textoParrafo=wrapper.find('p').text();
        //expect(wrapper).toMatchSnapshot();
        console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo);
    
    })
    
})