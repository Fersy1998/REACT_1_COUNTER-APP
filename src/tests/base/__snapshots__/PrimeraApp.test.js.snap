// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en la primera APp.js should say if App is lacking something new! "Hello, I´m Fersy" 1`] = `
<Fragment>
  <h1>
    <pre>
      Hello, I´m Fersy 23
    </pre>
  </h1>
  !!!
  <p>
    New One for third time
  </p>
</Fragment>
`;
